src/rgw/rgw_http_errors.h

```diff
@@ -10,65 +10,66 @@ struct rgw_http_errors {
-    { ERR_UNRESOLVABLE_EMAIL, 400, "UnresolvableGrantByEmailAddress" },
-    { ERR_INVALID_PART, 400, "InvalidPart" },
-    { ERR_INVALID_PART_ORDER, 400, "InvalidPartOrder" },
-    { ERR_REQUEST_TIMEOUT, 400, "RequestTimeout" },
-    { ERR_TOO_LARGE, 400, "EntityTooLarge" },
-    { ERR_TOO_SMALL, 400, "EntityTooSmall" },
-    { ERR_TOO_MANY_BUCKETS, 400, "TooManyBuckets" },
-    { ERR_MALFORMED_XML, 400, "MalformedXML" },
-    { ERR_AMZ_CONTENT_SHA256_MISMATCH, 400, "XAmzContentSHA256Mismatch" },
-    { ERR_LENGTH_REQUIRED, 411, "MissingContentLength" },
-    { EACCES, 403, "AccessDenied" },
-    { EPERM, 403, "AccessDenied" },
-    { ERR_SIGNATURE_NO_MATCH, 403, "SignatureDoesNotMatch" },
-    { ERR_INVALID_ACCESS_KEY, 403, "InvalidAccessKeyId" },
-    { ERR_USER_SUSPENDED, 403, "UserSuspended" },
-    { ERR_REQUEST_TIME_SKEWED, 403, "RequestTimeTooSkewed" },
-    { ERR_QUOTA_EXCEEDED, 403, "QuotaExceeded" },
-    { ENOENT, 404, "NoSuchKey" },
-    { ERR_NO_SUCH_BUCKET, 404, "NoSuchBucket" },
-    { ERR_NO_SUCH_WEBSITE_CONFIGURATION, 404, "NoSuchWebsiteConfiguration" },
-    { ERR_NO_SUCH_UPLOAD, 404, "NoSuchUpload" },
-    { ERR_NOT_FOUND, 404, "Not Found"},
-    { ERR_NO_SUCH_LC, 404, "NoSuchLifecycleConfiguration"},
-    { ERR_METHOD_NOT_ALLOWED, 405, "MethodNotAllowed" },
-    { ETIMEDOUT, 408, "RequestTimeout" },
-    { EEXIST, 409, "BucketAlreadyExists" },
-    { ERR_USER_EXIST, 409, "UserAlreadyExists" },
-    { ERR_EMAIL_EXIST, 409, "EmailExists" },
-    { ERR_KEY_EXIST, 409, "KeyExists"},
-    { ERR_INVALID_SECRET_KEY, 400, "InvalidSecretKey"},
-    { ERR_INVALID_KEY_TYPE, 400, "InvalidKeyType"},
-    { ERR_INVALID_CAP, 400, "InvalidCapability"},
-    { ERR_INVALID_TENANT_NAME, 400, "InvalidTenantName" },
-    { ENOTEMPTY, 409, "BucketNotEmpty" },
-    { ERR_PRECONDITION_FAILED, 412, "PreconditionFailed" },
-    { ERANGE, 416, "InvalidRange" },
-    { ERR_UNPROCESSABLE_ENTITY, 422, "UnprocessableEntity" },
-    { ERR_LOCKED, 423, "Locked" },
-    { ERR_INTERNAL_ERROR, 500, "InternalError" },
-    { ERR_NOT_IMPLEMENTED, 501, "NotImplemented" },
-    { ERR_SERVICE_UNAVAILABLE, 503, "ServiceUnavailable"}
+    { 0, 200, "", "" },
+    { STATUS_CREATED, 201, "Created", "Created" },
+    { STATUS_ACCEPTED, 202, "Accepted", "Accepted" },
+    { STATUS_NO_CONTENT, 204, "NoContent", "No content" },
+    { STATUS_PARTIAL_CONTENT, 206, "", ""  },
+    { ERR_PERMANENT_REDIRECT, 301, "PermanentRedirect", "The bucket you are attempting to access must be addressed using the specified endpoint. Send all future requests to this endpoint." },
+    { ERR_WEBSITE_REDIRECT, 301, "WebsiteRedirect", "Website redirect" },
+    { STATUS_REDIRECT, 303, "", "Status redirect" },
+    { ERR_NOT_MODIFIED, 304, "NotModified", "Not modified" },
+    { EINVAL, 400, "InvalidArgument", "Invalied Argument" },
+    { ERR_INVALID_REQUEST, 400, "InvalidRequest", "Please use AWS4-HMAC-SHA256; SOAP requests must be made over an HTTPS connection." },
```

Comment:
This string is not relevant to radosgw. We seem to currently return `InvalidRequest` in two places: as a fallback when we don't have a more specific error in a failure during bulk upload (https://github.com/ceph/ceph/blob/14867ad39887ad9a30511b257959fb00b46418ab/src/rgw/rgw_rest_swift.cc#L1519) and in `bulkdelete_respond` when the underlying error is unspecified, `ENOENT`, or `EACCES` (i.e. we didn't have a specific underlying error, or the request was to delete something that did not exist or to which permission was denied). 

I suggest changing this string to something generic like "Invalid Request"